### ISSUE TITLE
No symbols have been loaded for this document (TSX)

### ISSUE THREAD

**Issue by dfdumaresq** [microsoft/vscode-chrome-debug#985]
Issue Type: <b>Bug</b>

Set breakpoints on a .tsx file and launch the application. Breakpoints are not active. Debugger will stop on .cs files but not .tsx files.
This used to work.

Extension version: 4.12.6
VS Code version: Code 1.42.1 (c47d83b293181d9be64f27ff093689e8e7aed054, 2020-02-11T14:44:27.652Z)
OS version: Darwin x64 19.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|5, 4, 3|
|Memory (System)|16.00GB (0.07GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by roblourens** [comment]
Could you check whether this works with our new JS debugger? It should work much better: https://code.visualstudio.com/updates/v1_42#_new-javascript-debugger

**Comment by dfdumaresq** [comment]
I gave it a try but no luck, Rob. 
We recently upgraded our stack to ReactJS 16 and along the way changed our webpack.config.js slightly. 
To my eyes, it doesn't look important -- but I'll check to make sure it's picking up isDevBuild.
![image](https://user-images.githubusercontent.com/20051/75098135-63a56e80-5567-11ea-8147-c8366135bfc0.png)

**Comment by dfdumaresq** [comment]
Webpack seems to be outputting the correct files.
![image](https://user-images.githubusercontent.com/20051/75098427-f72c6e80-556a-11ea-8c1a-ab06165e82ef.png)

**Comment by dfdumaresq** [comment]
Here's a question: 
- In VSCode when I use cmd-p to open a file, I'm noticing invalid files. 
For example, I'm wanting to debug ApplyHome.tsx, but there are several listed under the path /var/...
Could these be confusing the debugger when attempting to locate the file which contains the breakpoint?
![image](https://user-images.githubusercontent.com/20051/75098713-6d7ea000-556e-11ea-9ba1-f33211afcdb0.png)

**Comment by roblourens** [comment]
Those files in "Recently Opened" are files that you opened at some point in vscode. If you try to open one and it doesn't exist, then it should get removed from the list. 

If you add a `debugger` statement to your code, does it stop there? Or could you set `"trace": true` (when using the new debug adapter) and upload the log file here? It prints the log file path to the debug console.

**Comment by dfdumaresq** [comment]
I'm able to do breakpointing in our previous release codebase. However, we just made a huge leap forward by upgrading the UI (among other pieces) from React 15 to React 16, now in development. So when I switch between them, I find the React 16 upgrade changed something which does not allow breakpointing.

**Comment by dfdumaresq** [comment]
I've created a dotnet/react app using the dotnet template and have the same issue. I can stop at breakpoints in .cs files, but not in .js files.
`dotnet new react -o hello-dotnet-react`

**Comment by dfdumaresq** [comment]
I'm going to

**Comment by dfdumaresq** [comment]
FYI - That didn't get me anywhere - it raised issues and requires netcoreapp3.x. We're using `netcoreapp2.1`.

**Comment by dfdumaresq** [comment]
@roblourens, I finally got the debugger to break - on an exception.
I've been noticing this error in the dotnet console:
`Exception thrown: 'System.IO.DirectoryNotFoundException' in System.Private.CoreLib.dll`

Which led me to turn on Breakpoints [x] Caught Exceptions
https://github.com/dotnet/runtime/issues/9123

And then the debugger kicked-in.
![image](https://user-images.githubusercontent.com/20051/75297487-dff8b580-57e4-11ea-8794-aa97db34994f.png)

This revealed that I'm generating debug logs now.
[vscode-debugadapter-13.json.zip](https://github.com/microsoft/vscode-chrome-debug/files/4252690/vscode-debugadapter-13.json.zip)

**Comment by dfdumaresq** [comment]
In vscode-debugadapter-13.json, I saw that huizhou.githd was causing problems, so I've disabled the extension.
I still don't have breakpoints working normally though... Here's an updated log file:
[vscode-debugadapter-0.json.zip](https://github.com/microsoft/vscode-chrome-debug/files/4252770/vscode-debugadapter-0.json.zip)

**Comment by roblourens** [comment]
Your sources are like `webpack:///./ClientApp/routes.tsx` so your sourceMapPathOverrides are not correct. If you remove it, the default one should kick in which should handle this. If not, you'll have to set one that maps a path like that to the correct location.

**Comment by dfdumaresq** [comment]
I'm confused about the launch settings. Should we use workspaceFolder or workspaceRoot? 
I'm now using
`    {
      "name": "Launch Chrome",
      "request": "launch",
      "type": "pwa-chrome",
      "url": "https://localhost:44301",
      "webRoot": "${workspaceFolder}/wwwroot",
      "sourceMaps": true,
      "trace": true
    },`

But still not getting any joy.
And have been getting error messages from VSCode saying it could not launch the executable (Chrome).
That seems to have been fixed by my adding the launch config above.
Adding the trace file
[vscode-debugadapter-1.json.zip](https://github.com/microsoft/vscode-chrome-debug/files/4257612/vscode-debugadapter-1.json.zip)

**Comment by roblourens** [comment]
Is this correct? `"webRoot": "${workspaceFolder}/wwwroot"`

**Comment by dfdumaresq** [comment]
Under wwwroot I have dist. 
So I'm now using  `"webRoot": "${workspaceRoot}/wwwroot/dist"` 
Does that make sense?

**Comment by dfdumaresq** [comment]
To answer my confusion about `workspaceRoot` vs `workspaceFolder` It seems `workspaceRoot` has been deprecated.
Good source: https://code.visualstudio.com/docs/editor/variables-reference

**Comment by roblourens** [comment]
webroot is kind of weird when using sourcemaps with webpack, the default sourceMapPathOverrides is actually like `webpack://?:*/*":"${webRoot}/*"` so webRoot should maybe be `${workspaceRoot}` in this case, which I think will be the default if you don't specify it.

**Comment by dfdumaresq** [comment]
Thanks for your reply, Rob!

I've tried a wide variety of launch settings, without any luck.
      // Attempting to Launch in develop - R16
       {
        "type": "chrome",
        "request": "launch",
        "name": "HTTPS Client",
        "url": "https://localhost:44301",
        "webRoot": "${workspaceRoot}/wwwroot",
        "sourceMaps": true,
        "trace": true,
       }

The debug console shows the paths are incorrectly based off root. Notice /ClientApp/...

![image](https://user-images.githubusercontent.com/20051/75405760-6d123c00-58c3-11ea-8f1a-73d7c618eac4.png)

In our previous release, before I had these problems a similar launch config shows paths correctly 
      {
        "type": "chrome",
        "request": "launch",
        "name": "HTTPS Client",
        "url": "https://localhost:44301",
        "trace": true,
        "webRoot": "${workspaceFolder}/wwwroot",
       }
![image](https://user-images.githubusercontent.com/20051/75406168-9e3f3c00-58c4-11ea-9d39-8fb1d2d52725.png)

Looking at the main-client.js.map for the R16 version, shows what look like garbage at the top.
[main-client.js.map.zip](https://github.com/microsoft/vscode-chrome-debug/files/4259186/main-client.js.map.zip)

The main-client-js.map for the R15 version shows correct paths as shown in the debug console.
[main-client.js.map.zip](https://github.com/microsoft/vscode-chrome-debug/files/4259191/main-client.js.map.zip)

I'm guessing we have work to do with our webpack.config.
Hope this helps.

**Comment by roblourens** [comment]
Not garbage - that's the actual sourcemap data but that sourcemap is missing the actual source paths so there must be a different sourcemap with those sources.

Did you try without webRoot like I suggested?

**Comment by dfdumaresq** [comment]
Thanks Rob. I have tried without webRoot.
       {
        "type": "chrome",
        "request": "launch",
        "name": "HTTPS Client",
        "url": "https://localhost:44301",
        // "webRoot": "${workspaceRoot}/",
        "sourceMaps": true,
        "trace": true,
       },

Chrome only shows the css source.
![image](https://user-images.githubusercontent.com/20051/75475545-d3dc3780-594d-11ea-93d4-6d0e99faf9f9.png)

HTTPS Client console:
![image](https://user-images.githubusercontent.com/20051/75475731-23bafe80-594e-11ea-9b1c-4e5834185774.png)

Dotnet console:
![image](https://user-images.githubusercontent.com/20051/75475629-f706e700-594d-11ea-9930-8e21862d1ede.png)

**Comment by dfdumaresq** [comment]
Pardon me, Rob. I am able to see the ClientApp directory structure and the transpiled source from within VSCode. 

![image](https://user-images.githubusercontent.com/20051/75477622-75b15380-5951-11ea-927a-8f1cecf2063e.png)

No breakpoints though.

I'm trying to locate or reproduce the sourceMaps you're suggesting may be elsewhere.

> sourcemap is missing the actual source paths so there must be a different sourcemap with those sources.

**Comment by dfdumaresq** [comment]
Success -- I have breakpoints working again!
One subtle change in webpack.config.js was to set `devtool: false`.
Also, my launch needed `"webRoot": "${workspaceFolder}/wwwroot"`

Thanks for all your help, Rob.

**Comment by dfdumaresq** [comment]
It leaves me with one question? 

How to get back to using the Debugger for Chrome extension. (Not all of my teammates are going to want to install the Javascript Debugger).

Now when I disable JD (nightly) and enable D4C, I'm given the error.
`Configured debug type 'pwa-chrome' is not supported.`

![image](https://user-images.githubusercontent.com/20051/75490263-b5d00080-5968-11ea-9db0-3144a5d5cc4f.png)

**Comment by dfdumaresq** [comment]
It leaves me with one question? 

How to get back to using the Debugger for Chrome extension. (Not all of my teammates are going to want to install the Javascript Debugger).

Now when I disable JD (nightly) and enable D4C, I'm given the error.
`Configured debug type 'pwa-chrome' is not supported.`

**Comment by dfdumaresq** [comment]
The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
[ddumaresq@ddumaresq epbc-ui] $ code --install-extension msjsdiag.debugger-for-chrome
Installing extensions...
Installing extension 'msjsdiag.debugger-for-chrome' v4.12.6...
Extension 'msjsdiag.debugger-for-chrome' v4.12.6 was successfully installed.
[ddumaresq@ddumaresq epbc-ui] $

**Comment by dfdumaresq** [comment]
Hi @roblourens - can you comment on how the above please, 
how to revert from Javascript Debugger to Debugger for Chrome?

**Comment by roblourens** [comment]
If you've set the `"debug.javascript.usePreview"` or `"debug.chrome.useV3"` settings, you need to unset those. But I would recommend using the new extension anyway as it will soon be the default builtin extension.

**Comment by roblourens** [comment]
> One subtle change in webpack.config.js was to set devtool: false.

That is strange because I think it is telling webpack to _not_ produce sourcemaps...

**Comment by dfdumaresq** [comment]
Yes, I agree. the `devtool: false` seems wrong. A red-herring so-to-speak. Still figuring out how to re-configure with webpack 4.

Also, I'm still trying to revert to Chrome but I'm not able to launch with type:"chrome". Still getting the `Configured debug type 'pwa-chrome' is not supported.`

My debug setting are:
- "debug.node.useV3": false
- "debug.extensionHost.useV3": false
- "debug.chrome.useV3": false

I only have this for usePreview 
- "search.useReplacePreview": true

**Comment by dfdumaresq** [comment]
@roblourens - an update:
I've put back the settings and am using the preview browser extension, Javascript Debugger. (Try try try, I could not return to the Chrome debugging extension. Not great. Meh.)

Anyway, I've found another issue that was resolved using webpack config `devtools: false`. 
[webpack-issue 8764](https://github.com/webpack/webpack/issues/8764) explains it is required when using the plugin, `webpack.SourceMapDevToolPlugin`


This helped me resolve the remaining pathing issue. It turns out my launch webroot did not specify the dist bundle directory.
This is working:
    {
      "type": "chrome",
      "request": "launch",
      "name": "HTTPS Client",
      "url": "https://localhost:44301",
      "webRoot": "${workspaceRoot}/wwwroot/dist",
    },

Feeling confident about closing this now. Thanks again for your help!
